### ISSUE TITLE
Button active and hover themes not working

### ISSUE THREAD

**Issue by MrBlenny** [grommet/grommet#4272]
It looks like button theming is not working correctly as of v2.14.0.
### Expected Behavior

Buttons should be themable by `button.active.primary.extend`,  `button.hover.primary.extend` etc as shown in [base.d.ts#L323](https://github.com/grommet/grommet/blob/master/src/js/themes/base.d.ts#L323). The custom button in the [storybook](https://storybook.grommet.io/?path=/story/button--custom) is not using this new theme format.

### Actual Behavior

When using the new format with `button.active.primary` instead of `button.primary.active` the theme has no effect.

**Comment by ShimiSun** [comment]
For the new themeing to apply you have to set button.default on your customer theme. You probably don't have it set so the props you mentioned may not be picked up by the theme.
https://v2.grommet.io/button
```
In May 2020, we enhanced Button to have three kinds, outlined below. In order to maintain backwards compatibility, this approach is only available when the theme being used defines button.default.
```

**Comment by MrBlenny** [comment]
Okay, I'll have a bit more of a play to see if there is some set of options I can give button.default that makes the buttons behave like the previous version.

As it stands, this is what happens when I set `button.default = {}` when no other button theming is applied.

![Screenshot from 2020-07-21 11-12-18@2x](https://user-images.githubusercontent.com/9685048/88001364-5bb41f80-cb43-11ea-88b8-de18a860bf0c.png)
![Screenshot from 2020-07-21 11-12-57@2x](https://user-images.githubusercontent.com/9685048/88001372-5ce54c80-cb43-11ea-91a7-079d1041c9e5.png)

Quite a drastic change in style when all I am using is a combination of custom colors and `primary` and `secondar` attributes on buttons.

**Comment by ShimiSun** [comment]
I agree, and for backward compatibility, you can always turn off the `button.default` on your theme.
If you do decide to opt-in the `button.default` behavior, you'll get more fine control on Button theme-ing, and you could use the `secondary` and `default` styling.

The Button enhancements we added on May (more theme capabilities, secondary and default buttons) would have been breaking changes for the previous Button behavior since the overall mindset is changed with the new `button.default` approach, hence we allowed to opt-in to the new Button features by setting up the flag of `button.default` on the theme that allowed us to maintain backward compatibility for the previous Button behavior.

We will improve the docs to indicate better which features are only included within the `button.default` and which aren't. I do think it is worth the investment of styling your theme with `button.default`, it will give you more button options and more styling capabilities as you scale your application.

**Comment by chancegraff** [comment]
Just started using Grommet and I'm having a terrible time attempting to get the button theming to work properly. Is there a better resource for understanding how this is supposed to function?

**Comment by taysea** [comment]
@chancegraff If you change your code to `<Button label="Where is my border" primary />` then the theming will apply as expected. When you provide the content within Button as children, it is treating the button as "plain", so no border is being applied. Typically, you will want to utilize the `label` and `icon` props for button contents.

**Comment by chancegraff** [comment]
Thank you so much for the information @taysea! I really appreciate it. Is this there somewhere I can read more about this in the docs? It would be super helpful to others in the future. I wasn't able to intuit that having direct descendants would flip the bit for the `plain` prop, and I suspect others would face the same issue moving forward. Making these available as composed elements would be amazing!

I realize it would be better to make a separate issue for this but I don't want to clog up the backlog with something that's written incorrectly. The submission process is honestly more than a little daunting, and I don't think I have all of the time needed to fill it out. If anyone sees this and wants to pick up my slack, you're free to take these suggestions and run with them!

**Comment by taysea** [comment]
Thanks for the feedback @chancegraff, I will read through the docs and find an appropriate spot to add it.

**Comment by ShimiSun** [comment]
Thanks @taysea , great work! Can we close this issue or is there any other improvements we can apply for it?

**Comment by taysea** [comment]
I think a pass at the docs to specify which theme objects are available only when `button.default` is defined would be beneficial. I can do a pass at that this afternoon.

**Comment by jcfilben** [comment]
I think this issue can be closed since #5025 was merged
